### PR TITLE
src/pam_python.c: Use new PyConfig API when building against Python 3…

### PR DESCRIPTION
**Pull request by sunweaver**
….12 any beyond.

### CODE REVIEW

File: src/pam_python.c [hunk 1]
```diff
@@ -127,6 +127,14 @@ static char libpython_so[]	= LIBPYTHON_SO;
 static void initialise_python(void)
 {
 #if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 204
+#if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 312
+  PyConfig config;
+  PyConfig_InitPythonConfig(&config);
+  config.write_bytecode = 1;
+  config.use_environment = 1;		/* Required to mitigate CVE-2019-16729 */
```
Comment (socketpair) [inline src/pam_python.c]: Seems you have ADDED a bug again. Here should be zeroes.

File: src/pam_python.c [hunk 2]
```diff
@@ -127,6 +127,14 @@ static char libpython_so[]	= LIBPYTHON_SO;
 static void initialise_python(void)
 {
 #if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 204
+#if	PY_MAJOR_VERSION*100 + PY_MINOR_VERSION >= 312
+  PyConfig config;
+  PyConfig_InitPythonConfig(&config);
```
Comment (socketpair) [inline src/pam_python.c:132]:
```cpp
PyConfig config;
PyConfig_InitIsolatedConfig(&config);   // note Isolated (!)!!!!!!
config.isolated = 1;
config.write_bytecode = 0;
config.use_environment = 0; /* Required to mitigate CVE-2019-16729 */
config.user_site_directory = 0; /* Required to mitigate CVE-2019-16729 */
config.site_import = 1;
config.install_signal_handlers = 0;
PyStatus status = Py_InitializeFromConfig(&config);
if (PyStatus_Exception(status)) {
   PyConfig_Clear(&config);
   if (!PyStatus_IsExit(status)) {
      Py_ExitStatusException(status);
   }
   abort(); // I don't know what to do here.
}
PyConfig_Clear(&config);
```

Comment (sunweaver) [reply src/pam_python.c:132]: Please revisit this MR. I have amended as suggested. Sorry for the long delay.

Comment (sunweaver) [reply src/pam_python.c:132]: @socketpair ^^^

**Comment by socketpair** [comment]
Oh, I did not see (lost notification)

**Pull request merged**